packages/frontend-2/components/project/model-page/dialog/embed/Embed.vue

```diff
@@ -0,0 +1,263 @@
+          <h4 class="font-bold text-sm text-foreground-2 mb-2 ml-0.5">Embed Code</h4>
+          <FormClipboardInput :value="iframeCode" is-multiline />
+          <p class="text-sm sm:text-base text-foreground-2 mt-2 mb-5 ml-0.5">
+            Copy this code to embed an iframe of model in your webpage or document.
+          </p>
+          <LayoutDialogSection border-b border-t title="Options">
+            <template #icon>
+              <Cog6ToothIcon class="h-full w-full" />
+            </template>
+            <div
+              class="flex flex-col gap-1.5 sm:gap-2 ml-5 sm:ml-7 text-sm cursor-default"
+            >
+              <div v-for="option in embedDialogOptions" :key="option.id">
+                <label
+                  :for="`option-${option.id}`"
+                  class="flex items-center gap-1 cursor-pointer max-w-max"
+                >
+                  <FormCheckbox
+                    :id="`option-${option.id}`"
+                    :model-value="option.value.value"
+                    :name="option.label"
+                    hide-label
+                    class="cursor-pointer"
+                    @update:model-value="
+                      (newValue) => updateOption(option.value, newValue)
+                    "
+                  />
+                  <span>{{ option.label }}</span>
+                </label>
+              </div>
+            </div>
+          </LayoutDialogSection>
+          <LayoutDialogSection
+            v-if="!isSmallerOrEqualSm"
+            border-b
+            title="Preview"
+            lazy-load
+            :lazy-load-height="400"
+          >
+            <template #icon>
+              <EyeIcon class="h-full w-full" />
+            </template>
+            <ProjectModelPageDialogEmbedIframe
+              v-if="!isSmallerOrEqualSm"
+              :src="updatedUrl"
+              title="Preview"
+              width="600"
+              height="400"
+              class="shrink-0 w-[600px] h-[400px] mx-auto"
+            />
+          </LayoutDialogSection>
+        </div>
+      </div>
+    </div>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { ref, computed } from 'vue'
+import { Cog6ToothIcon, EyeIcon } from '@heroicons/vue/24/outline'
+import { ProjectVisibility } from '~/lib/common/generated/gql/graphql'
+import { useClipboard } from '~~/composables/browser'
+import { SpeckleViewer } from '@speckle/shared'
+import { projectRoute } from '~~/lib/common/helpers/route'
+
+const props = defineProps<{
+  visibility?: ProjectVisibility
+  projectId: string
+  modelId?: string
+  versionId?: string
+}>()
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const router = useRouter()
+const route = useRoute()
+const { copy } = useClipboard()
+const {
+  public: { baseUrl }
+} = useRuntimeConfig()
+
+const { isSmallerOrEqualSm } = useIsSmallerOrEqualThanBreakpoint()
+
+const projectVisibility = ref(ProjectVisibility)
```

Comment:
You don't need to do this. As long as you have ProjectVisibility imported in the setup block, you can also use it in the template directly.